extensions/wikia/WikiFactory/api/ApiWikiFactorySaveVariable.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class ApiWikiFactorySaveVariable extends ApiBase {
+
+	/**
+	 * @throws UsageException
+	 */
+	public function execute() {
+		if ( !$this->getUser()->isAllowed( 'wikifactory' ) ) {
+			$this->dieUsageMsg( 'badaccess-groups' );
+		}
+
+		$params = $this->extractRequestParams();
+
+		try {
+			$result = WikiFactory::validateAndSetVarById(
+				$params['variable_id'],
+				$params['wiki_id'],
+				$params['variable_value'],
```

Comment:
Won't the variable value be always passed as a string (even if a user provides an array or an integer)?

Reply:
Yep it will be - the variable parser will take care of formatting based on the variable type as set in database.